### ISSUE TITLE
Passing --address and --internal-address while creating custom cluster is not set on the node.

### ISSUE THREAD

**Issue by ShylajaDevadiga** [rancher/rancher#33404]
Build from master-7a2fc74

Describe the issue:
--address and --internal-address is not stored. in config and therefore not set on the node. While passing --node-name is stored in config and set to the node.

Steps to reproduce:

- Click on Cluster Management->Create-Custom
- Enter Name to the the cluster-> Create
- Click on the name of the cluster in the next window
- Click on Show Advanced
- Pass Node-name, Node Public IP,  Node Private IP
- Copy the registration command and execute it on the node.
```
curl --insecure -fL https://<IP>/system-agent-install.sh | sudo  sh -s - --server https://<IP> --token <TOKEN> --ca-checksum <CHECKSUM> --etcd --controlplane --worker --address <Public IP> --internal-address <Private IP> --node-name shysrv1
```
Config yaml output
```
{
  "agent-token": "<TOKEN>",
  "cni": "calico",
  "etcd-snapshot-retention": 5,
  "etcd-snapshot-schedule-cron": "* */5 * * *",
  "node-label": [
    "rke.cattle.io/machine=e876f39c-7a92-4a2d-aa46-18e04862a88d"
  ],
  "node-name": "shysrv4",
  "server": "https://<SERVER PRIVATE IP>:9345",
  "token": "<TOKEN>"
}
```
Confirmed from the debug results, headers are correctly passed on correctly
```
X-Cattle-Node-Name: shysrv1 -H X-Cattle-Address: <PUBLIC IP> -H X-Cattle-Internal-Address: <PRIVATE IP>
```

**Other details that may be helpful:**
Node-name is set on the node as seen the kubectl output
```
kubectl get node
NAME      STATUS   ROLES                       AGE   VERSION
shysrv1   Ready    control-plane,etcd,master   86m   v1.21.2+rke2r1
```

**Environment information**
- Rancher version : 7a2fc74
- Installation option : Single/HA

**Cluster information**
- Cluster type : Custom
- Machine type : AWS

**Additional info:**
On DO InternalIP is set to Public IP
Describe the node
```
etcd.k3s.cattle.io/node-address: <Public IP>
InternalIP:  <Public IP>
```

**Comment by thedadams** [comment]
Leaving a note here: I have an implementation of this. I need to update the new provisioning tests when I return to complete this issue.

**Comment by catastrophe-brandon** [comment]
After running the registration command I see the following (`b395238`):

1. The specified node name shows up in the dashboard UI (Kube Node page). In this case, I used `my-node-name` as the node's name.
2. In the UI the external and internal IPs also have the correct values.
3. When I did a `kubectl get node` and the displayed name matches my expectations.
4. `kubectl describe node my-node-name` shows the correct name, internal IP, and external IP.

Based on all this, the issue has been fixed.
